### DISCUSSION TITLE
Again about the dying K2

### DISCUSSION THREAD

**Discussion by cmirnow** [eworkers/K2ForJ4#1] [General]
Hello.

I read your notice about unfinished work on githab, also read the discussion on joomlaworks.net. Tell me, are you sure that interrupting the work you started is the right decision? I don't see how a fork of software code under [GPL-2.0 license](https://github.com/getk2/k2/blob/master/LICENSE) could create "an unwanted inconvenience for some". In my opinion, it is impossible. You don't need my advice, but I'll give you my opinion: if you decide to continue your work, a huge number of Joomla users will be grateful to you.
Excuse my English.

Regards
Aleksei

**Comment by eworkers** [comment]
Hello @cmirnow 
Sorry for the late reply! but we were ridiculously overloaded with plenty of work!
To be true about the state of this repo, we think it is stable enough, but we don't want to advertise such announcement before we see some fair number of user implying so!

It will would be very kind of you if you clone and test and give us some feedback.
In following few days we will resume working on this repo to try to keep all users that are interested in only an intermediate version of k2, in a matter that they can get back again to k2 official version when it is ready.

Yet, yes we are planning to introduce a fork with refactoring involved (removing what does make sense anymore, introducing some new features and getting ready for joomla 5)

Meanwhile, we thank you for taking the time to introduce this discussion and we encourage all others to get involved so they can enlighten us more about next steps.

**Reply by cmirnow** [reply]
I had no problem upgrading from K2_v2.10.3 to K2ForJ4 (Joomla 3, php 8.0). In preparation for the upgrade Joomla certified two plugins as potentially incompatible with Joomla 4 - System K2 and User - K2 (potential upgrade issue). However, the upgrade from Joomla 3.10.11 to Joomla 4.2.5 went smoothly. This is cooler than I expected, really bravo.

I will continue testing soon. Thanks for your work, it gives me hope. In my opinion, regardless of the official release or not, it will be great to have a good fork of K2.

**Comment by PhilippSPOTTERON** [comment]
Hi, 
I have (as a graphic/UI designer) made a small logo extension for you for the files in media\k2\assets\images\backend - I do not have write access for upload or push. Could you enable that for my account so I can commit the images? Thanks- and thank you for the work on K2forJ4 - it is .. supercalifragilisticexpialidocious ;)

**Reply by eworkers** [reply]
Hi @PhilippSPOTTERON and thank you!
Pull requests are enabled for everyone on this repo, just change the image on a fork and make a pull request

**Comment by cmirnow** [comment]
@eworkers , would you like to fix comments for Joomla 3? 
https://github.com/eworkers/K2ForJ4/issues/35
This is a critical bug. Imho.
